### ISSUE TITLE
WebGL background & clear odd behaviour

### ISSUE THREAD

**Issue by JetStarBlues** [processing/p5.js#5195]
#### Most appropriate sub-area of p5.js?

- [x] WebGL

#### Details about the bug:

- p5.js version: 1.2.0, 1.3.1 (I suspect this bug is not version specific)

#### Description of the bug:

When you try to clear the screen (with [`clear()`][0] or [`background()`][1]), a weird subtle thing happens.

Consider this snippet (live version [here][2]):

```js
function setup()
{
  createCanvas(400, 400, WEBGL);

  noFill();
  stroke( 200, 255, 200 );  // lime
  strokeWeight( 1 );
}

function draw()
{
  // clear();  // equivalent-ish to `background( 0, 0, 0, 0 )`
  // background( 0, 0, 0, 0 );  // black w/ alpha zero
  background( 255, 0, 0, 0 );  // red w/ alpha zero
  // background( 255, 255, 255, 0 );  // white w/ alpha zero

  scale( 4 );
  box();

  noLoop();
}
```
It should draw a box stroked with a lime-green color. However, in addition to the lime-green stroke, a ghost-like stroke appears in some regions that corresponds to the color used in `background()` or `clear()`*. Here are some examples:

- `clear` or black background with alpha zero `background( 0, 0, 0, 0 )`  
![bug_stroke+blackBackground](https://user-images.githubusercontent.com/4354703/115313128-f91e9d00-a12f-11eb-9c1e-054bda8a6e89.png)

- detail of `clear` or black background with alpha zero `background( 0, 0, 0, 0 )`  
![bug_stroke+blackBackground_detail](https://user-images.githubusercontent.com/4354703/115313298-3daa3880-a130-11eb-8eee-6b614c678e6d.png)

- red background with alpha zero `background( 255, 0, 0, 0 )`  
![bug_stroke+redBackground](https://user-images.githubusercontent.com/4354703/115313330-4bf85480-a130-11eb-8eac-e9ea3df48d68.png)

- detail of red background with alpha zero `background( 255, 0, 0, 0 )`  
![bug_stroke+redBackground_detail](https://user-images.githubusercontent.com/4354703/115313340-51559f00-a130-11eb-8752-bfba77661d2d.png)

- white background with alpha zero `background( 255, 255, 255, 0 )`  
![bug_stroke+whiteBackground](https://user-images.githubusercontent.com/4354703/115313370-5fa3bb00-a130-11eb-9677-8a231d69ea1a.png)

- detail of white background with alpha zero `background( 255, 255, 255, 0 )`. A red background has been added with an image editor for contrast.  
![bug_stroke+whiteBackground_detail](https://user-images.githubusercontent.com/4354703/115313388-68948c80-a130-11eb-91dc-e743e2f4468e.png)

The effect does not seem to be unique to the stroke shader. The same effect can be observed with a filled box.

The bug seems to be triggered by using a color with an alpha that is not equal to `1` as an argument to `this.GL.clearColor` ([clear implementation][3], [background implementation][4]).

*Even though `clear()` does not take a color parameter, [behind the scenes][5], the color (0,0,0,0) is used.

[0]: https://p5js.org/reference/#/p5/clear
[1]: https://p5js.org/reference/#/p5/background
[2]: https://editor.p5js.org/acedean/sketches/QQeavzoXo
[3]: https://github.com/processing/p5.js/blob/a9f384fd79c616b9dab5414948b511791b62d344/src/webgl/p5.RendererGL.js#L896
[4]: https://github.com/processing/p5.js/blob/a9f384fd79c616b9dab5414948b511791b62d344/src/webgl/p5.RendererGL.js#L586
[5]: https://github.com/processing/p5.js/blob/a9f384fd79c616b9dab5414948b511791b62d344/src/webgl/p5.RendererGL.js#L901

**Comment by stalgiag** [comment]
Hi @JetStarBlues thanks for the thorough issue!

I think this is due to subpixel rendering or antialiasing. If you look at your example in Chrome with `setAttributes('antialias', false)`, you should no longer see the artifacts. Whereas, this will have no impact in Firefox which will display the artifact at certain zoom levels but not others. The last example you posted seems the most correct to me with no hue being added to the blending with the dom element in the background.

I hesitate to say this is caused by p5 in specific.

**Comment by JetStarBlues** [comment]
Thank you for pinpointing the issue!

I guess I'm expecting that the color used for the anti-aliasing will correspond to the color of the object. For example in this case, the lime green would be used, but varying in opacity, to generate the anti-aliasing. And that the background color will play no role whatsoever.

**Comment by davepagurek** [comment]
Hi! I think this should be fixed now as of #5917, as the weird blending with the background seems to only happen when premultiplied alpha is turned off. I'm going to close this for organizational purposes but feel free to discuss more or ask for this to be reopened if you notice anything weird still going on in the next p5 release!
